pkg/policy/repository.go

```diff
@@ -369,3 +373,85 @@ func (p *Repository) NumRules() int {
 func (p *Repository) GetRevision() uint64 {
 	return p.revision
 }
+
+// ConvertToK8sServiceToToCIDR traverses all egress rules and matches them against provided serviceInfo. If a matching egress rule is found it is populated with ToCIDR and ToPorts entries based on endpoint object.
+func (p *Repository) ConvertToK8sServiceToToCIDR(serviceInfo types.K8sServiceNamespace, endpoint types.K8sServiceEndpoint) error {
+	for _, rule := range p.rules {
+		for index, egress := range rule.Egress {
+			for _, service := range egress.ToServices {
+				// TODO: match services by labels
+				if service.K8sService == serviceInfo {
+					if err := generateToCidrFromEndpoint(&rule.Egress[index], endpoint); err != nil {
+						return err
+					}
+					if err := generateToPortsFromEndpoint(&rule.Egress[index], endpoint); err != nil {
+						return err
+					}
+				}
+			}
+		}
+	}
+	return nil
+}
+
+// generateToCidrFromEndpoint takes an egress rule and populates it with ToCIDR rules based on provided enpoint object
+func generateToCidrFromEndpoint(egress *api.EgressRule, endpoint types.K8sServiceEndpoint) error {
+	for ip := range endpoint.BEIPs {
+		epIP := net.ParseIP(ip)
+		// TODO: this will only work for IPv4. How to retrieve the mask from IPv6 address?
+		mask := epIP.DefaultMask()
+
+		found := false
+		for _, c := range egress.ToCIDR {
+			_, cidr, err := net.ParseCIDR(string(c))
+			if err != nil {
+				return err
+			}
+			if cidr.Contains(epIP) {
+				found = true
+				break
+			}
+		}
+		if !found {
+			cidr := net.IPNet{IP: epIP.Mask(mask), Mask: mask}
+			egress.ToCIDR = append(egress.ToCIDR, api.CIDR(cidr.String()))
```

Comment:
What worries me a bit is that by looking at the policy rule, it is not clear what part of `ToCIDR` is specified by the user and what part has been automatically populated. I see multiple options:

 - We add another rule which contains the automatically generated values. This keeps user-specified values completely separate from the automatically generated part.
 - We add separate CIDR and Ports fields for automatically generated values and keep it in the same rule.

Reply:
Isn't the fact that we are only manipulating `ToCIDR` rules within egress rules which have `ToServices` defined enough?

Reply:
A rule could have both, `ToCIDR` and `ToServices`. How do you know if you do `GET /policy` whether a CIDR value has been provided manually or has been populated?

Reply:
I'm also fine to disallow the combination of `ToCIDR` and `ToServices` but I would prefer if we have as few ways as possible for invalid policy to be specified.

Reply:
I think disallowing the combination would make sense from semantic POV, but since we don't give users immediate feedback for submitting wrong policies I agree with the sentiment.

Could this be done as another PR, as I feel that this PR is already getting quite big?

Reply:
I'm fine with that as long as we know and agree on the desired end state.